### PR TITLE
add index requests panel to reads/writes dashboard

### PR DESCRIPTION
**Pull request by sandeepsukhani**
**What this PR does / why we need it**:
In PR #7154, I refactored the code also to measure the latency on index write requests. 
This PR adds the index requests panel to the Loki Reads/Writes dashboard.

### CODE REVIEW

File: production/loki-mixin-compiled/dashboards/loki-reads.json [hunk 1]
```diff
@@ -630,7 +630,7 @@
                   "steppedLine": false,
                   "targets": [
                      {
-                        "expr": "sum by (status) (\n  label_replace(label_replace(rate(loki_boltdb_shipper_request_duration_seconds_count{cluster=~\"$cluster\",job=~\"($namespace)/(querier|index-gateway)\", operation=\"Shipper.Query\"}[$__rate_interval]),\n  \"status\", \"${1}xx\", \"status_code\", \"([0-9])..\"),\n  \"status\", \"${1}\", \"status_code\", \"([a-z]+)\"))\n",
+                        "expr": "sum by (status) (\n  label_replace(label_replace(rate(index_request_duration_seconds_count{cluster=~\"$cluster\",job=~\"($namespace)/querier\", operation!=\"put_chunk\"}[$__rate_interval]),\n  \"status\", \"${1}xx\", \"status_code\", \"([0-9])..\"),\n  \"status\", \"${1}\", \"status_code\", \"([a-z]+)\"))\n",
```
Comment (dannykopping) [inline production/loki-mixin-compiled/dashboards/loki-reads.json:633]: Do these metrics not have a `loki_` prefix?
If not, can we add it please to be consistent with our other metrics?

**Review by dannykopping — COMMENTED** [review]
LGTM, but the metric introduced by the previous PR (I didn't have time to review it before it was merged) seems like it doesn't have the `loki_` prefix.

File: production/loki-mixin-compiled/dashboards/loki-reads.json (hunk 1 above)
Comment (dannykopping) [reply production/loki-mixin-compiled/dashboards/loki-reads.json:633]: Oh, looking at the previous PR I don't see the metric - my bad for making an assumption.

Comment (owen-d) [inline production/loki-mixin-compiled/dashboards/loki-reads.json:633]: Do these work? It looks like the actual metrics have a `loki_prefix`:

```go
			Namespace: "loki",
			Name:      "index_request_duration_seconds",
```

**Review by owen-d — APPROVED** [review]
Nit, but LGTM

File: pkg/storage/store.go [hunk 2]
```diff
@@ -217,7 +217,7 @@ func (s *store) storeForPeriod(p config.PeriodConfig, chunkClient client.Client,
 			}
 			idx := series.NewIndexGatewayClientStore(gw, nil)
 
-			return failingChunkWriter{}, idx, func() {
+			return failingChunkWriter{}, index.NewMonitoredReaderWriter(idx, indexClientReg), func() {
```
Comment (sandeepsukhani) [inline pkg/storage/store.go:220]: I had missed wrapping the index reader/writer here so I fixed it.

**Comment by sandeepsukhani** [comment]
Fixed the metric prefix and the label filter for chunk flush operation

**Pull request merged**